### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by har777**
Refer: https://github.com/NethermindEth/cairo-vm-go/issues/413

### CODE REVIEW

File: integration_tests/cairo_files/test.small.cairo [hunk 1]
```diff
@@ -1,6 +1,44 @@
-%builtins range_check
+%builtins range_check bitwise
```
Comment (TAdev0) [inline integration_tests/cairo_files/test.small.cairo:1]: how can use use layout small for this file ? Bitwise is not in the small layout right?

File: integration_tests/cairo_files/blake.small.cairo [hunk 2]
```diff
@@ -0,0 +1,124 @@
+// inspired from the blake integration tests in the lambdaclass cairo-vm codebase
+
+%builtins range_check bitwise
```
Comment (TAdev0) [inline integration_tests/cairo_files/blake.small.cairo:3]: how can use use layout small for this file ? Bitwise is not in the small layout right?

File: integration_tests/cairo_files/test.small.cairo (hunk 1 above)
Comment (har777) [reply integration_tests/cairo_files/test.small.cairo]: you can't. PR is in draft :p

File: pkg/vm/builtins/poseidon.go [hunk 3]
```diff
@@ -57,7 +57,7 @@ func (p *Poseidon) GetAllocatedSize(segmentUsedSize uint64, vmCurrentStep uint64
 	if err != nil {
 		return 0, err
 	}
-	return allocatedInstances * cellsPerPedersen, nil
+	return allocatedInstances * cellsPerPoseidon, nil
```
Comment (TAdev0) [inline pkg/vm/builtins/poseidon.go]: haha indeed its better like this

File: pkg/vm/builtins/builtin_runner.go [hunk 4]
```diff
@@ -54,5 +54,5 @@ func GetAllocatedInstances(ratio uint64, cellsPerInstance uint64, segmentUsedSiz
 		return 0, fmt.Errorf("Number of steps must be at least %d. Current step: %d", minSteps, vmCurrentStep)
 	}
 	value := vmCurrentStep / ratio
-	return value * cellsPerInstance, nil
+	return value, nil
```
Comment (TAdev0) [inline pkg/vm/builtins/builtin_runner.go]: why would you remove cellsPerInstance here?

File: pkg/vm/builtins/layouts.go [hunk 5]
```diff
@@ -37,7 +37,7 @@ func getPlainLayout() Layout {
 }
 
 func getStarknetWithKeccakLayout() Layout {
-	return Layout{Name: "starknet_with_keccak", RcUnits: 16, Builtins: []LayoutBuiltin{
+	return Layout{Name: "starknet_with_keccak", RcUnits: 4, Builtins: []LayoutBuiltin{
```
Comment (TAdev0) [inline pkg/vm/builtins/layouts.go:40]: why 4 Rcunits? Its 16 for plain and small no?

Comment (har777) [reply pkg/vm/builtins/layouts.go:40]: from the python vm: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/lang/instances.py#L292

File: pkg/vm/builtins/builtin_runner.go (hunk 4 above)
Comment (har777) [reply pkg/vm/builtins/builtin_runner.go]: ah thats an outdated diff. was debugging some stuff.

File: pkg/vm/builtins/layouts.go (hunk 5 above)
Comment (TAdev0) [reply pkg/vm/builtins/layouts.go:40]: oh yeah indeed. Thats like a black box for me but lets go aha

**Review by TAdev0 — APPROVED** [review]

File: pkg/hintrunner/zero/zerohint_dictionaries.go [hunk 6]
```diff
@@ -478,7 +478,7 @@ func newSquashDictHint(dictAccesses, ptrDiff, nAccesses, bigKeys, firstKey hinte
 			//> ids.first_key = key = keys.pop()
 
 			//> address = ids.dict_accesses.address_
-			address, err := dictAccesses.GetAddress(vm)
+			address, err := hinter.ResolveAsAddress(vm, dictAccesses)
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_dictionaries.go:481]: yep ! indeed!

File: pkg/hintrunner/zero/zerohint_dictionaries.go [hunk 7]
```diff
@@ -714,12 +714,14 @@ func newSquashDictInnerCheckAccessIndexHint(loopTemps hinter.ResOperander) hinte
 }
 
 func createSquashDictInnerCheckAccessIndexHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
-	loopTemps, err := resolver.GetResOperander("loop_temps")
+	loopTemps, err := resolver.GetCellRefer("loop_temps")
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_dictionaries.go:717]: oh yeah.. is there any way to guess its a cellRefer just with the python code? i dont think so

File: pkg/hintrunner/zero/zerohint_math_test.go [hunk 8]
```diff
@@ -390,48 +390,48 @@ func TestZeroHintMath(t *testing.T) {
...
+					{Name: "output", Kind: fpRelative, Value: addr(8)},
 					{Name: "value", Kind: fpRelative, Value: feltInt64(15)},
 					{Name: "base", Kind: fpRelative, Value: feltInt64(2)},
 					{Name: "bound", Kind: fpRelative, Value: feltInt64(5)},
 				},
 				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
 					return newSplitIntHint(ctx.operanders["output"], ctx.operanders["value"], ctx.operanders["base"], ctx.operanders["bound"])
 				},
-				check: varValueEquals("output", feltInt64(1)),
+				check: consecutiveVarAddrResolvedValueEquals("output", []*fp.Element{feltInt64(1)}),
```
Comment (TAdev0) [inline pkg/hintrunner/zero/zerohint_math_test.go:406]: why is this change needed?

File: pkg/runners/zero/zero.go [hunk 9]
```diff
@@ -303,11 +303,17 @@ func (runner *ZeroRunner) checkRangeCheckUsage() error {
 	var rcUnitsUsedByBuiltins uint64
 	for _, builtin := range runner.program.Builtins {
 		if builtin == starknet.RangeCheck {
-			bRunner := builtins.Runner(builtin)
-			rangeCheckRunner, _ := bRunner.(*builtins.RangeCheck)
-			rangeCheckSegment, ok := runner.vm.Memory.FindSegmentWithBuiltin(rangeCheckRunner.String())
-			if ok {
-				rcUnitsUsedByBuiltins += rangeCheckSegment.Len() * rangeCheckRunner.RangeCheckNParts
+			for _, layoutBuiltin := range runner.layout.Builtins {
```
Comment (TAdev0) [inline pkg/runners/zero/zero.go:306]: nice indeed!

**Review by TAdev0 — APPROVED** [review]

File: pkg/runners/zero/zero.go (hunk 9 above)
Comment (TAdev0) [reply pkg/runners/zero/zero.go:306]: what was the bug related to this? 

i just see that a file with rangecheck usage. but with a layout plain could successfully execute this function?

Comment (har777) [reply pkg/runners/zero/zero.go:306]: Only faced an issue with the blake hash test.

File: pkg/hintrunner/zero/zerohint_math_test.go (hunk 8 above)
Comment (har777) [reply pkg/hintrunner/zero/zerohint_math_test.go:406]: ah we need a var addr resolved value not a var value

Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_math_test.go:406]: oh yeah indeed, thanks

File: pkg/runners/zero/zero.go (hunk 9 above)
Comment (TAdev0) [reply pkg/runners/zero/zero.go:306]: oh ok
